birdhouse/components/proxy/default.env

```diff
@@ -6,23 +6,31 @@ export PROXY_IMAGE="nginx:1.23.4"
 # Any WPS processes taking longer than this should use async mode.
 export PROXY_READ_TIMEOUT_VALUE="240s"
 
+export BIRDHOUSE_PROXY_SCHEME='$([ x"$BIRDHOUSE_HTTP_ONLY" = x"True" ] && echo http || echo https )'
+export BIRDHOUSE_ALLOW_UNSECURE_HTTP='$([ x"$BIRDHOUSE_HTTP_ONLY" = x"True" ] && echo True || echo )'
+export PROXY_INCLUDE_HTTPS='$([ x"$BIRDHOUSE_HTTP_ONLY" = x"True" ] || echo "include /etc/nginx/conf.d/https.include;" )'
+
 # Content of "location /" in file config/proxy/conf.d/all-services.include.template
 # Useful to have a custom homepage.
 # Note that the default homepage will become the jupyterhub login page if the jupyterhub component is enabled.
 # If the jupyterhub component is not enabled, it is highly recommended to create a custom homepage since the magpie
 # landing page is not the most user-friendly option.
-export BIRDHOUSE_PROXY_ROOT_LOCATION="return 302 https://\$host/jupyter/hub/login;"
+export BIRDHOUSE_PROXY_ROOT_LOCATION='return 302 ${BIRDHOUSE_PROXY_SCHEME}://\$host/jupyter/hub/login;'
 
-export INCLUDE_FOR_PORT_80='$([ x"$BIRDHOUSE_ALLOW_UNSECURE_HTTP" = x"True" ] && echo "include /etc/nginx/conf.d/all-services.include;" || echo "include /etc/nginx/conf.d/redirect-to-https.include;")'
+export PROXY_INCLUDE_FOR_PORT_80='$([ x"$BIRDHOUSE_ALLOW_UNSECURE_HTTP" = x"True" ] && echo "include /etc/nginx/conf.d/all-services.include;" || echo "include /etc/nginx/conf.d/redirect-to-https.include;")'
```

Comment:
This change here is only changing the variable name to follow the naming convention introduced in 2.4.0. This doesn't need to be in the PR but as long as we're fixing things I brought the variable name up to date.

> Granted, the BIRDHOUSE_PROXY_SCHEME updates would be needed to report the http rather than https where it was harcoded

Yes, this is the important change.

> but it seems the https.include rewrite is not needed (the HTTPS server definition is just ignored).

It's needed if you don't have a self signed SSL certificate. Otherwise the proxy container won't start since nginx fails to parse a valid SSL certificate in the https block.

Reply:
Sorry. My comment wasn't clear.
The variable rename is fine for alignment with 2.x, and so is the `BIRDHOUSE_PROXY_SCHEME` addition.

What I was referring to is the contents of the variable. I placed the comment here because I found it better illustrated the `BIRDHOUSE_ALLOW_UNSECURE_HTTP=True` I was referring to.

Agreed, the SSL certificate file is needed for `nginx` config to resolve.
I think that if you use the `BIRDHOUSE_ALLOW_UNSECURE_HTTP=True` approach, you don't even need the contents to be an actual SSL cert file. It just needs to be "something" for the volume mount to work (but I'm not sure, to validate).

Might be worth checking out if the following works:
```bash
touch ssl-dont-care.pem

BIRDHOUSE_ALLOW_UNSECURE_HTTP=True \
BIRDHOUSE_SSL_CERTIFICATE="$(realpath ssl-dont-care.pem)" \
	birdhouse compose up -d 
```

Otherwise, the self-signed certificate described in https://github.com/bird-house/birdhouse-deploy/pull/484#issuecomment-2521037780 can be used to have a more "legit" file.

Reply:
Oh I see...
Yes, it has to be a well-formatted certificate file or else nginx complains and won't start.